ctk/common/src/main/kotlin/org/codice/compliance/utils/TestCommon.kt

```diff
@@ -245,29 +279,137 @@ class TestCommon {
+        fun loginAndGetCookies(binding: SamlProtocol.Binding, relayState: String? = null,
+            singleSP: Boolean = true): Map<String, String> {
+            try {
+
+                val authnRequest = createDefaultAuthnRequest(binding)
+                val secondRequest =
+                    createDefaultAuthnRequest(binding, DSA_SP_ISSUER, DSA_SP_ENTITY_INFO)
+
+                return if (binding == HTTP_POST) {
+                    val firstLoginResponse = loginPost(authnRequest, relayState)
+                    if (!singleSP) loginPost(secondRequest, relayState)
+                    firstLoginResponse.cookies
+                } else {
+                    val firstLoginResponse = loginRedirect(authnRequest, relayState)
+                    if (!singleSP) loginRedirect(secondRequest, relayState)
+                    firstLoginResponse.cookies
+                }
+            } catch (e: Exception) {
+                throw SAMLComplianceException.create(SAMLGeneral_d,
+                    message = "The logout test is unable to run because an error occurred while " +
+                        "logging in.",
+                    cause = e)
+            }
+        }
+
+        private fun loginPost(request: AuthnRequest, relayState: String? = null): Response {
+            val response = sendPostAuthnRequest(
+                signAndEncodeToString(request, relayState))
+            verifyHttpStatusCode(response.statusCode)
+            return getServiceProvider(IdpSSOResponder::class).getResponseForPostRequest(response)
+        }
+
+        private fun loginRedirect(request: AuthnRequest, relayState: String? = null): Response {
+            val queryParams = SimpleSign().signUriString(
+                SAML_REQUEST,
+                encodeRedirectRequest(request),
+                relayState)
+
+            val response = sendRedirectAuthnRequest(queryParams)
+            verifyHttpStatusCode(response.statusCode)
+            return getServiceProvider(IdpSSOResponder::class)
+                .getResponseForRedirectRequest(response)
+        }
+
+        /**
+         * Provides a default logout request for testing
+         * @return A valid LogoutRequest.
+         */
+        fun createDefaultLogoutRequest(binding: SamlProtocol.Binding): LogoutRequest {
+            REQUEST_ID = UUID.randomUUID().toString().replace("-", "")
+            return LogoutRequestBuilder().buildObject().apply {
+                issuer = IssuerBuilder().buildObject().apply { value = currentSPIssuer }
+                id = REQUEST_ID
+                version = SAMLVersion.VERSION_20
+                issueInstant = DateTime()
+                destination = getSingleLogoutLocation(binding.uri)
+                nameID = NameIDBuilder().buildObject().apply {
+                    nameQualifier = idpMetadata.entityId
+                    spNameQualifier = currentSPIssuer
+                    format = PERSISTENT_ID
+                    value = "admin"
```

Comment:
Currently, just trying to log out of the admin user. My goal is to have a properties file or something of that sorts where implementers can set the test user username and password.